### PR TITLE
[needs revision] Embed Augeas Httpd lens

### PR DESCRIPTION
**Pull request by SwartzCr**
add httpd.aug (from v1.4) to the letsencrypt-apache directory and load it when initializing augeas

### CODE REVIEW

**Comment by SwartzCr** [comment]
deals with issue #981

**Comment by kuba** [comment]
Travis is all red :(

**Comment by SwartzCr** [comment]
I know. None of the tests fail locally, I'm trying to debug :/

File: letsencrypt-apache/letsencrypt_apache/augeas_configurator.py [hunk 1]
```diff
@@ -30,7 +33,8 @@ def __init__(self, *args, **kwargs):
         # Set Augeas flags to not save backup (we do it ourselves)
         # Set Augeas to not load anything by default
         my_flags = augeas.Augeas.NONE | augeas.Augeas.NO_MODL_AUTOLOAD
-        self.aug = augeas.Augeas(flags=my_flags)
+        my_dir = os.path.dirname(constants.AUGEAS_HTTPD_LENS)
```
Comment (kuba) [inline letsencrypt-apache/letsencrypt_apache/augeas_configurator.py]: I'm not sure if it's safe to call `dirname` on the `resource_filename` result... `resource_filename` might create a temporary file (copy it from the distribution to some location on the disk), especially when `zip_safe == True`

Comment (kuba) [inline letsencrypt-apache/letsencrypt_apache/augeas_configurator.py]: It seems like Augeas really needs directory. https://pythonhosted.org/setuptools/pkg_resources.html#resource-extraction: "If the named resource is a directory, then all resources within that directory (including subdirectories) are also extracted". So, please move the lens to a subdirectory (`letsencrypt-apache/letsencrypt_apache/augeas_lenses/httpd.aug`), and use `AUGEAS_LENSES = pkg_resources.resource_filename('letsencrypt_apache', 'lenses')`

Comment (kuba) [inline letsencrypt-apache/letsencrypt_apache/augeas_configurator.py]: also, please leave a comment about it somewhere in the repo, maybe `letsencrypt-apache/letsencrypt_apache/augeas_lenses/README`?

**Comment by SwartzCr** [comment]
I'm still blocked on this, testing it as a local branch here: https://github.com/SwartzCr/letsencrypt/pull/3
@kuba do you have any ideas?

**Comment by pde** [comment]
I can confirm that the unit tests pass for me locally.

**Comment by kuba** [comment]
I just spun up DO droplet with 12.04 x64. It fails with the same errors. It installs `libauges0==0.10.0-0ubuntu4`, so I guess that is the problem.

**Comment by kuba** [comment]
@SwartzCr, if you still plan on working on this PR, I updated it to use the "right" `pkg_resources` commands at: https://github.com/kuba/letsencrypt/tree/embed-lens. I also merged with `master` - you branched from a veryyyyy old commit - that could cause a lot of problems, so try not to do it in the feature and merge before with `master` without debugging further.

**Comment by kuba** [comment]
https://gist.github.com/kuba/e873180b96de276c04c1

File: letsencrypt-apache/letsencrypt_apache/augeas_configurator.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 """Class of Augeas Configurators."""
 import logging
+import os
```
Comment (pde) [inline letsencrypt-apache/letsencrypt_apache/augeas_configurator.py]: This import is causing the lint error; os isn't used after kuba's commits.

**Comment by kuba** [comment]
Does it work on precise? It's not clear, because of the move to trusty (#1358)...

**Comment by pde** [comment]
The apache plugin doesn't work with it on precise, but the client should be okay.  Will double check before merging.

**Comment by kuba** [comment]
So are we effectively dropping support for Ubuntu 12.04 Long Term Support? That does not sound good :(

**Comment by domcleal** [comment]
Precise appears to have Augeas 0.10.0, so that new lens from 1.4.0 won't work with that version of the library.  There was a change/fix to the square lenses in Augeas 1.0.0 so certain newer lenses (Httpd being one) can't be run on 0.x.

**Comment by pde** [comment]
@kuba People on Ubuntu 12.04 will need to use standalone or webroot for now. That's not ideal, but it's better than not being able to [handle the stock config in 15.04+]https://github.com/letsencrypt/letsencrypt/issues/981()

**Comment by pde** [comment]
Did some extra testing.  This seems solid on platforms where augeas 1.2+ is available, and seems to degrade gracefully (no apache plugin) on older ones.

**Pull request merged**